packages/backend/src/tests/propertyBased/arbitraries/rbitraryWhere.ts

```diff
@@ -0,0 +1,80 @@
+import { fc } from '@fast-check/vitest';
+import { AllTablesRowsMap } from '../properties/executeQuery';
+import { arbitraryWhereValue } from './arbitraryWhereValue';
+import { Schema, Where, getTableWhereableColumns } from '@synthql/queries';
+import { AnyDb } from '../../../types';
+
+export function arbitraryWhere<DB>({
+    schema,
+    allTablesRowsMap,
+    tableName,
+    validWhere,
+}: {
+    schema: Schema<DB>;
+    allTablesRowsMap: AllTablesRowsMap;
+    tableName: string;
+    validWhere: boolean;
+}): fc.Arbitrary<Where<AnyDb, string>> {
+    return fc
+        .constantFrom(...getTableWhereableColumns(schema, tableName))
+        .chain((columnName) => {
+            // TODO: We should remove this check once we resolve the `timestampz` issue
+            if (
+                checkIfTimestampzColumn({
+                    allTablesRowsMap,
+                    tableName,
+                    columnName,
+                })
+            ) {
+                return arbitraryWhere({
+                    schema,
+                    allTablesRowsMap,
+                    tableName,
+                    validWhere,
+                });
+            } else {
+                return fc.dictionary(
+                    fc.constant(columnName),
+                    arbitraryWhereValue({
+                        allTablesRowsMap,
+                        tableName,
+                        columnName,
+                        validWhere,
+                    }),
+                    {
+                        depthIdentifier: '0',
+                        minKeys: 1,
+                        maxKeys: 1,
+                    },
+                );
+            }
+        });
+}
+
+function checkIfTimestampzColumn({
```

Comment:
I'm a bit confused about the purpose of this function.

If it is to figure out if a column has the type `timestampz` this is not a good/reliable check. You should just get the PG type from the schema. This should be really easy to add to the schema.